client/signup/steps/plans/index.jsx

```diff
@@ -33,25 +33,25 @@ export class PlansStep extends Component {
 		experimentLoaded: false,
 	};
 
-	componentWillMount() {
+	componentDidMount() {
+		this.unsubscribe = subscribeIsDesktop( ( matchesDesktop ) =>
+			this.setState( { isDesktop: matchesDesktop } )
+		);
+		this.props.saveSignupStep( { stepName: this.props.stepName } );
+
 		if ( this.props.flowName === 'onboarding' || this.props.flowName === 'launch-site' ) {
-			loadExperimentAssignment( 'calypso_signup_monthly_plans_default_202201_v1' ).then(
+			loadExperimentAssignment( 'calypso_signup_monthly_plans_default_202201_v2' ).then(
 				( experiment ) => {
+					// eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
This is okay to do the job, but an ideal solution would be to avoid doing a `setState()` on mount. The loading of the experiment could happen in another component that's responsible only for that. One way to do that would be to implement an experiment assignment loader component, which would work in a similar way that a query component does.